### PR TITLE
Fix many documentation errors in libcudf.

### PR DESCRIPTION
**Pull request by karthikeyann**
This PR fixes many documentation errors in libcudf code. (include and src)
Many errors are found through doxygen warnings.

### CODE REVIEW

File: cpp/include/cudf/io/csv.hpp [hunk 1]
```diff
@@ -603,7 +603,7 @@ class csv_reader_options {
   /**
    * @brief Sets indexes of columns to read as datetime.
    *
-   * @param col_names Vector of column indices to infer as datetime.
+   * @param col_ind Vector of column indices to infer as datetime.
    */
   void set_parse_dates(std::vector<int> col_ind) { _parse_dates_indexes = std::move(col_ind); }
```
Comment (davidwendt) [inline cpp/include/cudf/io/csv.hpp:608]:
```suggestion
   * @param col_indices Vector of column indices to infer as datetime.
   */
  void set_parse_dates(std::vector<int> col_indices) { _parse_dates_indexes = std::move(col_indices); }
```

File: cpp/include/cudf/io/csv.hpp [hunk 2]
```diff
@@ -620,7 +620,7 @@ class csv_reader_options {
   /**
    * @brief Sets indexes of columns to parse as hexadecimal
    *
-   * @param col_names Vector of column indices to parse as hexadecimal
+   * @param col_ind Vector of column indices to parse as hexadecimal
    */
   void set_parse_hex(std::vector<int> col_ind) { _parse_hex_indexes = std::move(col_ind); }
```
Comment (davidwendt) [inline cpp/include/cudf/io/csv.hpp:625]:
```suggestion
   * @param col_indices Vector of column indices to parse as hexadecimal
   */
  void set_parse_hex(std::vector<int> col_indices) { _parse_hex_indexes = std::move(col_indices); }
```

File: cpp/include/cudf/rolling/range_window_bounds.hpp [hunk 3]
```diff
@@ -38,15 +38,15 @@ struct range_window_bounds {
   /**
    * @brief Factory method to construct a bounded window boundary.
    *
-   * @param value Finite window boundary
+   * @param scalar_ Finite window boundary
    *
    */
-  static range_window_bounds get(scalar const&);
+  static range_window_bounds get(scalar const& scalar_);
```
Comment (davidwendt) [inline cpp/include/cudf/rolling/range_window_bounds.hpp:44]: How about `boundary` instead of `scalar_` ?

File: cpp/include/cudf/sorting.hpp [hunk 4]
```diff
@@ -81,7 +81,7 @@ std::unique_ptr<column> stable_sorted_order(
  * @brief Checks whether the rows of a `table` are sorted in a lexicographical
  *        order.
  *
- * @param[in] in                table whose rows need to be compared for ordering
+ * @param[in] table                table whose rows need to be compared for ordering
```
Comment (davidwendt) [inline cpp/include/cudf/sorting.hpp:84]:
```suggestion
 * @param[in] table             Table whose rows need to be compared for ordering
```

File: cpp/include/cudf/rolling/range_window_bounds.hpp (hunk 3 above)
Comment (karthikeyann) [reply cpp/include/cudf/rolling/range_window_bounds.hpp:44]: updated

**Review by davidwendt — APPROVED** [review]
Thanks for doing this.

**Review by harrism — APPROVED** [review]
Thanks for doing this. Does this mean we are close to being able to run doxygen in CI to catch these errors?

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**